include/swift/Runtime/Mutex.h

```diff
@@ -20,12 +20,14 @@
 
 #include <type_traits>
 
-#if (defined(__APPLE__) || defined(__linux__) || defined(__CYGWIN__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__))
+#ifdef SWIFT_STDLIB_SINGLETHREADED_RUNTIME
+#include "swift/Runtime/MutexSingleThreaded.h"
+#elif (defined(__APPLE__) || defined(__linux__) || defined(__CYGWIN__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__))
```

Comment:
Not your doing, but I think that using `_POSIX_THREADS` is a better check for this.

Reply:
https://github.com/apple/swift/pull/33609